doc/website/scripts/clone_module.py

```
@@ -0,0 +1,42 @@
+from pathlib import Path
+
+
+def clone_external_module(url: str, ref: str) -> tuple[Path, tempfile.TemporaryDirectory]:
+    """Shallow-clone url@ref into a temp dir.
+
+    Returns (clone_path, tmp_dir). Caller must call tmp_dir.cleanup() when done.
+    Raises subprocess.CalledProcessError on failure — caller is responsible for
+    catching and skipping the module gracefully.
+    """
+    tmp = tempfile.TemporaryDirectory(prefix="holohub_module_")
+    subprocess.run(
```

Ruff (0.15.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)